update retrieval quality article

thierrypdamiba opened the pull request:
Make changes to the retrieval quality article

Commit:
update retreival quality article

davidmyriel commented:
@thierrypdamiba `client.query_points ` has this been changed in client version 1.12? Doesnt work for me

joein commented on qdrant-landing/content/documentation/tutorials/retrieval-quality.md:
```diff
@@ -1,227 +1,158 @@
-[Loading a dataset from Hugging Face hub](/documentation/tutorials/huggingface-datasets/) tutorial, `Qdrant/arxiv-titles-instructorxl-embeddings`
-from the [Hugging Face hub](https://huggingface.co/datasets/Qdrant/arxiv-titles-instructorxl-embeddings). Let's download it in a streaming
-mode, as we are only going to use part of it.
+We’ll use a pre-embedded dataset from Hugging Face to train and test Qdrant’s search capabilities. First, load and split the dataset for training (1,000 items) and testing (100 items). 
```
differs from the code values

joein commented:
@thierrypdamiba @davidmyriel I actually liked the fact that in the previous version we said that embeddings quality is crucial (maybe we paid it a bit more attention than required) and we explained why we're comparing exact search to ann, now the tutorial has become a bit faceless

Commit:
added info about embedding quality and ann vs exact search

thierrypdamiba commented:
@joein @davidmyriel  I added information about the quality and ann vs exact search. Also updated the numbers on the dataset to reflect the code.

Commit:
added information for context of the process while keeping it in steps

generall commented on qdrant-landing/package.json:
```diff
@@ -21,7 +21,8 @@
     "clipboard": "^2.0.11",
-    "qdrant-page-search": "^1.0.8"
+    "qdrant-page-search": "^1.0.8",
+    "react-router-dom": "^6.27.0"
```
why do we need this?

thierrypdamiba replied:
We don't need it. Removing now.

Commit:
Update package.json to remove unnecessary react-router-dom package

Commit:
Update package.json to fix formatting

Commit:
Update retrieval-quality.md content

Update text and format to better reflect the benefit of ANN vs KNN/exact search and why a user would want to measure retrieval quality"

TODO: Add screenshots of how you can do this in the webui

Commit:
Update retrieval-quality.md to fix formatting

Commit:
add webui images and explanations

Commit:
fix image links

Commit:
link to article data for images

joein commented on qdrant-landing/content/documentation/tutorials/retrieval-quality.md:
```diff
@@ -117,111 +91,126 @@ client.upload_points(  # upload_points is available as of qdrant-client v1.7.1
+- **m**: This parameter determines the maximum number of connections per node in the HNSW graph. A higher value for `m` increases the connectivity of the graph, potentially improving search accuracy at the cost of increased memory usage and indexing time. The default value for `m` is 16.
+- **ef_construct**: This parameter controls the size of the dynamic candidate list during index construction. A higher value of `ef_construct` leads to a more exhaustive search during the indexing phase, resulting in a higher quality graph and improved search accuracy. However, this comes at the cost of longer indexing times. The default value for `ef_construct` is 100.
+
+We will use the untuned HNSW as the baseline to compare how changes affect the precision of the search. Initially, we will use the default values of `m` (16) and `ef_construct` (100) for the HNSW algorithm. Later, we will double these values to observe their impact on retrieval quality.
```
We have already written what the default values are, so we can shorten this sentence, like 
"We'll use the default `m` and `ef` as a baseline and then tweak the params to see how it affects the precision of the search."

joein commented on qdrant-landing/content/documentation/tutorials/retrieval-quality.md:
```diff
@@ -117,111 +91,126 @@ client.upload_points(  # upload_points is available as of qdrant-client v1.7.1
+### How to Select HNSW Parameters
 
-Response:
+- If you require higher precision, increase `m` and `ef_construct` while considering the increased memory usage and indexing time.
+- If memory and indexing time are critical constraints, tune the parameters incrementally to find the right balance.
```
By the way, these is also a third parameter : `ef` (also known as `efSearch`, it controls the number of neighbors evaluated during the search, a higher value may increase precision, however, it also increases latency

joein commented on qdrant-landing/package.json:
```diff
@@ -33,4 +33,4 @@
   "name": "qdrant-landing",
   "version": "0.1.0"
-}
+}
```
nit: changes in package.json and package-lock.json can be completely discarded

Commit:
remove double reference to default ef_construct and m and add ef search content

joein commented on qdrant-landing/content/documentation/tutorials/retrieval-quality.md:
````diff
@@ -117,111 +91,126 @@ client.upload_points(  # upload_points is available as of qdrant-client v1.7.1
 ```
 
-Response:
+This step measures the initial retrieval quality before any tuning of the HNSW parameters. The HNSW (Hierarchical Navigable Small World) algorithm has two key parameters that influence search performance and quality:
````
We could provide a bit more details here:
There are 2 types of parameters which users can tune, index time parameters and search time parameters
index time: `m` and `ef_construct`, search time - `ef`

I think that we might want to mention it here, rather than just add a brief sentence at the end of the article
However, I don't find the code adjustments to be a necessity

Commit:
add details about HNSW parameters to search quality tutorial

Commit:
Minor formatting updates to parameter descriptions in retrieval-quality.md
